main.py

```diff
@@ -55,95 +54,86 @@ def main():
-                        '-to', out_point,
-                        '-i', input_file,
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-vf', 'yadif',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750',                        
-                        '-c:a', 'aac',                        
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                    
+            ffmpeg_command = create_ffmpeg_command(input_file)
+
+            print("Transcoding with:", " ".join(ffmpeg_command), "\n")
+
+            subprocess.call(ffmpeg_command)
+
+
+def create_ffmpeg_command(input_file):
+    output_path = set_output_path(input_file)
+
+    to_watermark = set_watermark_status()
+    to_trim = set_trim_status()
+
+    aspect_ratio = stream_aspect_ratio(input_file)
+    filtergraph = build_filtergraph(aspect_ratio, to_watermark)
+
+    ffmpeg_program_call = [
+        "ffmpeg"
+    ]
+
+    input_video_file = [
+        "-i", input_file
+    ]
+
+    general_stream_settings = [
+        "-c:v",     "libx264",          # encode video stream as H.264
+        "-pix_fmt", "yuv420p",          # 4:2:0 chroma subsubsampling
+        "-c:a",     "aac",              # encode audio stream(s) as AAC
+        "-map",     "0",       "-dn"    # map all streams except data streams to output
+    ]
+
+    output_settings = [
+        "-metadata", "copyright=Media Archive for Central England",
+        "-metadata", "comment=Please contact MACE to license this footage on 01522 837750",
+        output_path,
+        "-report"     # Generates log in current directory
+    ]
+
+    if not to_trim:
+        return ffmpeg_program_call + input_video_file + filtergraph + general_stream_settings + output_settings
+
+    # If the user selects to trim...
+    elif to_trim:
+        # Get the start and end trim in points from the user.
+        in_point = set_timestamp(
+            "Please specify the trim 'in' point")
+        out_point = set_timestamp(
+            "Please specify the trim 'out' point")
+
+        trim_settings = [
+            "-ss", in_point,
+            "-to", out_point
+        ]
+
+        return ffmpeg_program_call + trim_settings + input_video_file + filtergraph + stream_settings + output_settings
+
+
+def build_filtergraph(aspect_ratio, to_watermark):
+    dar = display_aspect_ratio(aspect_ratio)
+    watermark_file = watermark_path(aspect_ratio)
+
+    general_filtergraph = [
+        "-vf", "setdar=dar={0}".format(dar) # set DAR (display aspect ratio) to aspect ratio
+    ]
+
+    watermark_settings = [
+        "-i",               watermark_file,
+        "-filter_complex", "setdar=dar={0},overlay".format(dar)
```

Comment:
Previously the command was `"[0]yadif=0:-1:1[v];[v][1]overlay"`. Presumably there is a similar/more sophisticated way to include both `setdar` and `overlay` in the filtergraph, but this seems to work, and I believe in keeping it simple where possible :laughing: